### ISSUE TITLE
Recommend check / test version numbering

### ISSUE THREAD

**Issue by jarrodmillman** [scientific-python/upload-nightly-action#41]
See https://github.com/matplotlib/matplotlib/issues/26736#issuecomment-1714417050

Currently I am not able to figure out how to easily install the recent nightly wheels for matplotlib due to version numbering issues and upper pins.

I wonder if we could automate testing whether it is possible to install the nightly wheels using the recommended pip command:
```
python -m pip install \
  --upgrade \
  --pre \
  --index-url https://pypi.anaconda.org/scientific-python-nightly-wheels/simple \
  --extra-index-url https://pypi.org/simple \
  matplotlib
```

**Comment by jarrodmillman** [comment]
Just so folks don't need to follow the link above:

```
>>> from packaging import version
>>> version.parse("3.8.0rc1") < version.parse("3.8.0.dev1951+g5f785e3e19")
False
>>> version.parse("3.8.0rc1") < version.parse("3.8.0rc1.dev1951+g5f785e3e19")
False
>>> version.parse("3.8.0rc1") < version.parse("3.8.0rc1.dev0")
False
>>> version.parse("3.8.0rc1") < version.parse("3.8.0rc2.dev0")
True
>>> version.parse("3.8.0rc1") < version.parse("3.8.0rc2.dev1951+g5f785e3e19")
True
```

So after releasing `3.8.0rc1` the basic version number should be updated to `3.8.0rc2.dev0` or `3.8.0rc2.dev`.

**Comment by jarrodmillman** [comment]
> As for testing, would you suggest to test whether it's possible to install the latest uploaded nightly?

Yes, exactly. Maybe a cron job once a week or so to test whether the recommended pip command is able to install the most recent nightly wheels.

**Comment by martinfleis** [comment]
And what would be the action if the job fails? Getting back to a project, suggesting they need to change the version of the dev branch?

**Comment by bsipocz** [comment]
> And what would be the action if the job fails? Getting back to a project, suggesting they need to change the version of the dev branch?

Yes, I suppose an issue being auto-opened here, and then one of us following up with the project. This of course assumes that the problem is scaling well, e.g. we won't host gazillions of wheels as a "conda-forge for wheels", but keep it to the core ones.

**Comment by matthewfeickert** [comment]
I didn't follow this closeley in 2023 (sorry), but given @ksunden's comment https://github.com/matplotlib/matplotlib/issues/26736#issuecomment-1714548480 was this actually a systematic problem that a check would have help avoid? Or was this simply a time delay issue for information to get propaged to `matplotlib`'s `main` branch so it could end up in the `matplotlib` nightly wheels?
